currencies/rate_converter.go

```diff
@@ -111,6 +116,9 @@ func (rc *RateConverter) Update() error {
 		rc.rates.Store(rates)
 		rc.lastUpdated.Store(time.Now())
 	} else {
+		if rc.CheckStaleRates() {
+			rc.rates.Store((*Rates)(nil))
+		}
 		glog.Errorf("Error updating conversion rates: %v", err)
```

Comment:
Ok, let me check if I've got this right. If there's an error fetching the rates json and we should check for stale rates (CheckStaleRates returns true), set the rates to nil which is a special marker to tell it to update next time?